test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ControllerTest.cs

```diff
@@ -1797,34 +1779,41 @@ public void TempData_CanSetAndGetValues()
             Assert.Equal(input, result);
         }
 
-        private static Controller GetController(IModelBinder binder, IValueProvider provider)
+        private static Controller GetController(IModelBinder binder, IValueProvider valueProvider)
         {
             var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
             var httpContext = new DefaultHttpContext();
-            var actionContext = new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
 
             var viewData = new ViewDataDictionary(metadataProvider, new ModelStateDictionary());
             var tempData = new TempDataDictionary(Mock.Of<IHttpContextAccessor>(), Mock.Of<ITempDataProvider>());
 
-            var bindingContext = new ActionBindingContext()
+            var controllerContext = new ControllerContext()
             {
-                ModelBinder = binder,
-                ValueProvider = provider,
+                HttpContext = httpContext,
                 InputFormatters = new List<IInputFormatter>(),
-                ValidatorProvider = new DataAnnotationsModelValidatorProvider(
-                    options: null,
-                    stringLocalizerFactory: null)
+                ModelBinders = new List<IModelBinder>()
```

Comment:
Just test code, but would this work:

```c#
var ct = new CT()
{
    ...
    ModelBinders = new[] { binder },
    ValueProviders = new[] { valueProvider },
};
```

Etc.?

Is the compiler smart enough to figure all the covariant stuff out?